From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sven Strickroth <email@cs-ware.de>
Date: Thu, 1 Aug 2024 21:53:44 +0200
Subject: Allow to reset and free refstore

Signed-off-by: Sven Strickroth <email@cs-ware.de>
---
 refs.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/refs.c b/refs.c
index d3caa9a633..e60c64ed90 100644
--- a/refs.c
+++ b/refs.c
@@ -2343,6 +2343,14 @@ static struct ref_store *ref_store_init(struct repository *repo,
 	return refs;
 }
 
+void ref_store_release_and_clear(struct repository *repo)
+{
+	if (!repo->refs_private)
+		return;
+	ref_store_release(repo->refs_private);
+	FREE_AND_NULL(repo->refs_private);
+}
+
 void ref_store_release(struct ref_store *ref_store)
 {
 	ref_store->be->release(ref_store);
