From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sven Strickroth <email@cs-ware.de>
Date: Fri, 14 Aug 2015 12:53:14 +0200
Subject: Do not print commits on STDOUT

Signed-off-by: Sven Strickroth <email@cs-ware.de>
---
 log-tree.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/log-tree.c b/log-tree.c
index 4590384812..b5000b9bbb 100644
--- a/log-tree.c
+++ b/log-tree.c
@@ -946,7 +946,9 @@ int log_tree_diff_flush(struct rev_info *opt)
 	}
 
 	if (opt->loginfo && !opt->no_commit_id) {
-		show_log(opt);
+		//show_log(opt);
+		opt->loginfo = NULL; // TGit: do not print commits on STDOUT and skip all calculations regarding formatting
+		opt->shown_one = 1;
 		if ((opt->diffopt.output_format & ~DIFF_FORMAT_NO_OUTPUT) &&
 		    opt->verbose_header &&
 		    opt->commit_format != CMIT_FMT_ONELINE &&
