From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sven Strickroth <email@cs-ware.de>
Date: Sun, 7 Feb 2016 16:20:24 +0100
Subject: Allow to schedule cleanup tasks

Signed-off-by: Sven Strickroth <email@cs-ware.de>
---
 run-command.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/run-command.c b/run-command.c
index 086e885ede..8fe6a49441 100644
--- a/run-command.c
+++ b/run-command.c
@@ -1107,7 +1107,7 @@ static struct {
 
 static int git_atexit_installed;
 
-static void git_atexit_dispatch(void)
+void git_atexit_dispatch(void)
 {
 	size_t i;
 
@@ -1115,7 +1115,7 @@ static void git_atexit_dispatch(void)
 		git_atexit_hdlrs.handlers[i-1]();
 }
 
-static void git_atexit_clear(void)
+void git_atexit_clear(void)
 {
 	free(git_atexit_hdlrs.handlers);
 	memset(&git_atexit_hdlrs, 0, sizeof(git_atexit_hdlrs));
