From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sven Strickroth <email@cs-ware.de>
Date: Tue, 4 Sep 2018 17:44:14 +0200
Subject: chdir notify needs cleanup

Signed-off-by: Sven Strickroth <email@cs-ware.de>
---
 chdir-notify.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/chdir-notify.c b/chdir-notify.c
index f8bfe3cbef..ec2318292b 100644
--- a/chdir-notify.c
+++ b/chdir-notify.c
@@ -97,6 +97,18 @@ int chdir_notify(const char *new_cwd)
 	return 0;
 }
 
+void cleanup_chdir_notify(void)
+{
+	struct list_head *pos;
+	struct list_head *p;
+	list_for_each_safe(pos, p, &chdir_notify_entries)
+	{
+		struct chdir_notify_entry *e = list_entry(pos, struct chdir_notify_entry, list);
+		free(e);
+	}
+	INIT_LIST_HEAD(&chdir_notify_entries);
+}
+
 char *reparent_relative_path(const char *old_cwd,
 			     const char *new_cwd,
 			     const char *path)
