From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sven Strickroth <email@cs-ware.de>
Date: Tue, 24 Dec 2019 14:07:58 +0100
Subject: Undefine ALTERNATE as it is already defined in <wingdi.h>

Signed-off-by: Sven Strickroth <email@cs-ware.de>
---
 fetch-pack.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fetch-pack.c b/fetch-pack.c
index a9fa07c2bb..094f7fc5ec 100644
--- a/fetch-pack.c
+++ b/fetch-pack.c
@@ -56,6 +56,7 @@ static struct strbuf fsck_msg_types = STRBUF_INIT;
 static struct string_list uri_protocols = STRING_LIST_INIT_DUP;
 
 /* Remember to update object flag allocation in object.h */
+#undef ALTERNATE // already defined in wingdi.h
 #define COMPLETE	(1U << 0)
 #define ALTERNATE	(1U << 1)
 #define COMMON		(1U << 6)
