From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sup Yut Sum <ch3cooli@gmail.com>
Date: Tue, 21 Jan 2020 08:22:53 +0800
Subject: Temporarily disable the use of commit-graph file

Signed-off-by: Sup Yut Sum <ch3cooli@gmail.com>
---
 commit-graph.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/commit-graph.c b/commit-graph.c
index 801471a098..30577f810b 100644
--- a/commit-graph.c
+++ b/commit-graph.c
@@ -243,6 +243,7 @@ static int commit_graph_compatible(struct repository *r)
 
 int open_commit_graph(const char *graph_file, int *fd, struct stat *st)
 {
+#if TGIT_ENABLE_COMMIT_GRAPH
 	*fd = git_open(graph_file);
 	if (*fd < 0)
 		return 0;
@@ -251,6 +252,9 @@ int open_commit_graph(const char *graph_file, int *fd, struct stat *st)
 		return 0;
 	}
 	return 1;
+#else
+	return 0;
+#endif
 }
 
 struct commit_graph *load_commit_graph_one_fd_st(struct odb_source *source,
