From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sven Strickroth <email@cs-ware.de>
Date: Thu, 18 Aug 2022 15:43:17 +0200
Subject: No need to load mailmap for showing the log

Signed-off-by: Sven Strickroth <email@cs-ware.de>
---
 builtin/log.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/builtin/log.c b/builtin/log.c
index 29931239aa..40a6e140ba 100644
--- a/builtin/log.c
+++ b/builtin/log.c
@@ -339,7 +339,7 @@ static void cmd_log_init_finish(int argc, const char **argv, const char *prefix,
 		rev->sources = &revision_sources;
 	}
 
-	if (mailmap) {
+	if (mailmap && 0) { /* don't load mailmap */
 		rev->mailmap = xmalloc(sizeof(struct string_list));
 		string_list_init_nodup(rev->mailmap);
 		read_mailmap(the_repository, rev->mailmap);
