From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sven Strickroth <email@cs-ware.de>
Date: Sat, 14 Feb 2026 21:25:48 +0100
Subject: Allow to close open pack files

Signed-off-by: Sven Strickroth <email@cs-ware.de>
---
 odb.c      | 9 +++++++++
 packfile.c | 7 +++++++
 2 files changed, 16 insertions(+)

diff --git a/odb.c b/odb.c
index 7d555be09f..67896954b2 100644
--- a/odb.c
+++ b/odb.c
@@ -1035,6 +1035,15 @@ void odb_close(struct object_database *o)
 	close_commit_graph(o);
 }
 
+void odb_close_tgit(struct object_database *o)
+{
+	clear_delta_base_cache();
+	odb_close(o);
+	struct odb_source *source;
+	for (source = o->sources; source; source = source->next)
+		packfile_store_close_tgit(source->packfiles);
+}
+
 static void odb_free_sources(struct object_database *o)
 {
 	while (o->sources) {
diff --git a/packfile.c b/packfile.c
index 7fbe47ca18..bc7dfb8fc8 100644
--- a/packfile.c
+++ b/packfile.c
@@ -2849,6 +2849,13 @@ void packfile_store_close(struct packfile_store *store)
 	store->midx = NULL;
 }
 
+void packfile_store_close_tgit(struct packfile_store *store)
+{
+	packfile_list_clear(&store->packs);
+	strmap_clear(&store->packs_by_path, 0);
+	store->initialized = false; // needed?
+}
+
 struct odb_packed_read_stream {
 	struct odb_read_stream base;
 	struct packed_git *pack;
