From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sven Strickroth <email@cs-ware.de>
Date: Mon, 22 Jun 2026 19:20:22 +0200
Subject: Drop support for Git < 2.47

Signed-off-by: Sven Strickroth <email@cs-ware.de>
---
 compat/mingw.c | 13 -------------
 path.c         |  2 +-
 2 files changed, 1 insertion(+), 14 deletions(-)

diff --git a/compat/mingw.c b/compat/mingw.c
index b2f295530a..65cf472131 100644
--- a/compat/mingw.c
+++ b/compat/mingw.c
@@ -4574,19 +4574,6 @@ int is_cygwin_msys2_hack_active(void)
 	return dwValue == 1;
 }
 
-int is_new_git_with_appdata(void)
-{
-	HKEY hKey;
-	DWORD dwValue = 0;
-	if (RegOpenKeyExW(HKEY_CURRENT_USER, L"Software\\TortoiseGit", 0, KEY_ALL_ACCESS, &hKey) == ERROR_SUCCESS) {
-		DWORD dwType = REG_DWORD;
-		DWORD dwSize = sizeof(dwValue);
-		RegQueryValueExW(hKey, L"git_cached_version", NULL, &dwType, (LPBYTE)&dwValue, &dwSize);
-		RegCloseKey(hKey);
-	}
-	return dwValue >= (2 << 24 | 46 << 16);
-}
-
 /*
  * Based on https://stackoverflow.com/questions/43002803
  *
diff --git a/path.c b/path.c
index b8616adaa0..a8506f3aa7 100644
--- a/path.c
+++ b/path.c
@@ -1597,7 +1597,7 @@ char *xdg_config_home_for(const char *subdir, const char *filename)
 		home_config = mkpathdup("%s/.config/%s/%s", home, subdir, filename);
 
 	#ifdef WIN32
-	if (!is_cygwin_msys2_hack_active() && is_new_git_with_appdata())
+	if (!is_cygwin_msys2_hack_active())
 	{
 		const char *appdata = getenv("APPDATA");
 		if (appdata && *appdata) {
